src_c/geometry.c

```diff
@@ -8,7 +8,77 @@
 
 #define PYGAMEAPI_GEOMETRY_NUMSLOTS 21
 
-static PyMethodDef _pg_module_methods[] = {{NULL, NULL, 0, NULL}};
+static PyObject *
+geometry_regular_polygon(PyObject *_null, PyObject *const *args,
+                         Py_ssize_t nargs)
+{
+    int sides;
+    double radius;
+    double angle = 0;
+    double Cx, Cy;
+
+    if (nargs < 3 || nargs > 4) {
+        return RAISE(PyExc_TypeError,
+                     "invalid number of arguments, expected 3 or 4 arguments");
+    }
+    if (!PyLong_Check(args[0])) {
+        return RAISE(PyExc_TypeError,
+                     "the first parameter must be an integer");
+    }
+    sides = PyLong_AsLong(args[0]);
+    if (sides == -1 && PyErr_Occurred()) {
+        return NULL;
+    }
```

Comment:
I think it's enough to do this because if `PyLong_AsLong` fails an error is automatically set and is the correct one anyway.
```suggestion
    sides = PyLong_AsLong(args[0]);
    if (PyErr_Occurred()) {
        return NULL;
    }
```

Reply:
Plus you can just check for PyErr_Occurred() because the error is always set when it fails, but it can happen that PyLong_AsLong returns -1 but it was intended to return -1.